### PR TITLE
Improve fgGetStaticFieldSeqAndAddress

### PR DESCRIPTION
**Pull request by EgorBo**
It turns out it's a bit too conservative by looking only at CNS_INT nodes and in a case I found I need to rely on VN instead, e.g.:

```
[000106] U--XG------                         \--*  IND       long   <l:$280, c:$281>
[000027] -------N---                            \--*  ADD       long   $141
[000026] ---------U-                               +--*  CAST      long <- ulong <- uint $100
[000025] -----------                               |  \--*  LSH       int    $46
[000098] -----------                               |     +--*  XOR       int    $44
[000097] -----------                               |     |  +--*  HWINTRINSIC int     LeadingZeroCount $42
[000096] -----------                               |     |  |  \--*  CNS_INT   int    445 $41
[000095] -----------                               |     |  \--*  CNS_INT   int    31 $43
[000024] -----------                               |     \--*  CNS_INT   int    3 $45
[000132] H----------                               \--*  CNS_INT(h) long   0x1b1b6542c18 static Fseq[..] $140
```
In this case `CAST` node has $100,$100 VN that represents an integer constant (with help of https://github.com/dotnet/runtime/pull/78929)

@dotnet/jit-contrib PTAL simple change.

### CODE REVIEW

File: 0001-Improve-GetStaticFieldSeqAndAddress.patch [hunk 1]
```diff
@@ -0,0 +1,63 @@
+From 68137c0aa93b554815760219db2110a70a31ecf9 Mon Sep 17 00:00:00 2001
```
Comment (TIHan) [inline 0001-Improve-GetStaticFieldSeqAndAddress.patch:1]: The .patch shouldn't be included right?

Comment (EgorBo) [reply 0001-Improve-GetStaticFieldSeqAndAddress.patch]: oops 🙂

**Review by TIHan — APPROVED** [review]
This looks like a simple and good change to me, pending CI.

Quick question, what is `GTF_ICON_STATIC_HDL` and why are we not including them?

**Comment by EgorBo** [comment]
> GTF_ICON_STATIC_HDL

This functions looks for trees where GTF_ICON_STATIC_HDL (address of a static field basically, in this case -- address of an RVA element) is blended with constant offsets, e.g.
ADD(ICON_STATIC, 10) or ADD(ADD(ICON_STATIC, 10), LCL) where LCL has "I am a constant" VN

**Comment by EgorBo** [comment]
Failure is https://github.com/dotnet/runtime/pull/78986 (fixed)

**Pull request merged**